components/tidb_query_datatype/src/codec/data_type/bit_vec.rs

```diff
@@ -73,6 +73,16 @@ impl BitVec {
         let pos = idx >> 3;
         (self.data[pos] & mask) != 0
     }
+
+    pub fn null_cnt(&self) -> usize {
+        let mut cnt = 0;
```

Comment:
I remember that there is a bit-wise operation to count zeros effectively. Can we use that?

Reply:
If you mean [HAKMEM](https://stackoverflow.com/questions/8590432/when-to-use-parallel-counting-mit-hakmem-for-bitcount-when-memory-is-an-issue) algorithm, I don't think there will make much difference, as it operates on `u64`. As we organize data in `u8` instead of `u64`, there might be memory align issues if we apply this algorithm.